src/server/streamableHttp.ts

```diff
@@ -0,0 +1,368 @@
+   * According to the MCP Streamable HTTP transport spec, the server MUST either return SSE or 405.
+   * We choose to return 405 Method Not Allowed as we don't support GET SSE connections yet.
+   */
+  private async handleGetRequest(req: IncomingMessage, res: ServerResponse): Promise<void> {
+    // Check session validity first for GET requests when session management is enabled
+    if (this._sessionId !== undefined && !this.validateSession(req, res)) {
+      return;
+    }
+
+    res.writeHead(405, {
+      ...this._customHeaders,
+      "Allow": "POST, DELETE"
+    }).end(JSON.stringify({
+      jsonrpc: "2.0",
+      error: {
+        code: -32000,
+        message: "Method not allowed: Server does not offer an SSE stream at this endpoint"
+      },
+      id: null
+    }));
+  }
+
+  /**
+   * Handles POST requests containing JSON-RPC messages
+   */
+  private async handlePostRequest(req: IncomingMessage, res: ServerResponse, parsedBody?: unknown): Promise<void> {
+    try {
+      // Validate the Accept header
+      const acceptHeader = req.headers.accept;
+      // The client MUST include an Accept header, listing both application/json and text/event-stream as supported content types.
+      if (!acceptHeader ||
+        !acceptHeader.includes("application/json") || !acceptHeader.includes("text/event-stream")) {
+        res.writeHead(406).end(JSON.stringify({
+          jsonrpc: "2.0",
+          error: {
+            code: -32000,
+            message: "Not Acceptable: Client must accept both application/json and text/event-stream"
+          },
+          id: null
+        }));
+        return;
+      }
+
+      const ct = req.headers["content-type"];
+      if (!ct || !ct.includes("application/json")) {
+        res.writeHead(415).end(JSON.stringify({
+          jsonrpc: "2.0",
+          error: {
+            code: -32000,
+            message: "Unsupported Media Type: Content-Type must be application/json"
+          },
+          id: null
+        }));
+        return;
+      }
+
+      let rawMessage;
+      if (parsedBody !== undefined) {
+        rawMessage = parsedBody;
+      } else {
+        const parsedCt = contentType.parse(ct);
+        const body = await getRawBody(req, {
+          limit: MAXIMUM_MESSAGE_SIZE,
+          encoding: parsedCt.parameters.charset ?? "utf-8",
+        });
+        rawMessage = JSON.parse(body.toString());
+      }
+
+      let messages: JSONRPCMessage[];
+
+      // handle batch and single messages
+      if (Array.isArray(rawMessage)) {
+        messages = rawMessage.map(msg => JSONRPCMessageSchema.parse(msg));
+      } else {
+        messages = [JSONRPCMessageSchema.parse(rawMessage)];
+      }
+
+      // Check if this is an initialization request
+      // https://spec.modelcontextprotocol.io/specification/2025-03-26/basic/lifecycle/
+      const isInitializationRequest = messages.some(
+        msg => 'method' in msg && msg.method === 'initialize'
+      );
```

Comment:
An initialization request is not allowed to be batched with anything else, so we should enforce that there's only one message if initializing.